src/Core/Waiter.php

```diff
@@ -121,20 +128,39 @@ public function __destruct()
      */
     final public function resolve(?float $timeout = null): bool
     {
-        if (!isset($this->response)) {
+        if (null !== $this->resolveResult) {
+            if ($this->resolveResult instanceof \Exception) {
+                throw $this->resolveResult;
+            }
+
+            if (\is_bool($this->resolveResult)) {
+                return $this->resolveResult;
+            }
+        }
+
+        if (null === $this->response) {
             return true;
         }
 
         try {
-            if (null !== $timeout && $this->httpClient->stream($this->response, $timeout)->current()->isTimeout()) {
-                return false;
+            if (null !== $timeout) {
```

Comment:
We don’t need this If statement, right?

The behavior will be the same without it.

Reply:
We don't need it. But test won't pass because MockClient::stream does accept SimpleMockResponse

Reply:
removed (and fixed issue with MockStream)